### PR TITLE
Find morphology location from a coordinate

### LINKED ISSUE #1661: Feature: create locset from 3d points
For a given cell morphology, located in space, we want to find the location on the morphology that is closest to a 3d location.

Two possible approaches:
1. simple: iterate over all segments via `pwlin`, testing to find the location of the cell morphology that is closest to the location
2. batch: pre process the morphology to create a data structure that allows for fast lookup of locations in less than `O(n_seg)` time. Useful if lookup of many locations on the same cell morphology.

### PR DESCRIPTION
**Pull request by bcumming**
Extend the `pwlin_place` interface to find the location of a morphology that is closest to a 3d coordinate.
* extend `arb::pwlin_place` interface
* python wrapper
* unit tests
* documentation

Addresses a feature request by @Helveg that I don't think ever had a ticket assigned to it.

Fixes #1661 and #1108.

### CODE REVIEW

**Comment by noraabiakar** [comment]
I think this solves #1661?

**Comment by noraabiakar** [comment]
Also, I just merged #1719, which may cause some issues because it touched `embed_pwlin` as well.

File: test/unit/test_morph_place.cpp [hunk 1]
```diff
@@ -570,3 +570,104 @@ TEST(place_pwlin, segments) {
...
+    segment_tree tree;
+
+    //  the test morphology:
+    //
+    //       x=-6        x=6
+    //
+    //         _        _
+    //          \       / y=35
+    //      seg4 \     / seg3
+    //  branch 2  \   \     branch 1
```
Comment (noraabiakar) [inline test/unit/test_morph_place.cpp:584]: typo: `// branch 2 \ / branch 1`

File: test/unit/test_morph_place.cpp [hunk 2]
```diff
@@ -570,3 +570,104 @@ TEST(place_pwlin, segments) {
...
+    //             \ / seg2
+    //              | y=20
```
Comment (noraabiakar) [inline test/unit/test_morph_place.cpp:586]: typo: y = 25

File: test/unit/test_morph_place.cpp [hunk 3]
```diff
@@ -570,3 +570,104 @@ TEST(place_pwlin, segments) {
     EXPECT_TRUE(mpoint_almost_eq(p8p, x3all[2].prox));
     EXPECT_TRUE(mpoint_almost_eq(p8p, x3all[2].dist));
 }
+
+TEST(place_pwlin, nearest) {
+    segment_tree tree;
+
+    //  the test morphology:
+    //
+    //       x=-6        x=6
```
Comment (noraabiakar) [inline test/unit/test_morph_place.cpp:579]: typo: `x = -9   x = 9`

**Review by noraabiakar — CHANGES_REQUESTED** [review]
lgtm! Only issue is that the representation of the morphology in the unit tests doesn't match the the segment tree used.

File: test/unit/test_morph_place.cpp (hunk 1 above)
Comment (bcumming) [reply test/unit/test_morph_place.cpp:584]: I made a proper mess of the diagram.

File: test/unit/test_morph_place.cpp (hunk 3 above)
Comment (bcumming) [reply test/unit/test_morph_place.cpp:579]: Fixed.

**Review by noraabiakar — APPROVED** [review]

**Pull request merged**